pkg/config/value_constraints.go

```diff
@@ -0,0 +1,39 @@
+package config
+
+type DistributionVersion struct {
+	Name  string
+	Label string
+}
+type DistributionArch struct {
+	Name  string
+	Label string
+}
```

Comment:
You might ask, why are these structs defined in both places (api and config).  The reason is that the values defined need to be read by both the api and the model layer (for validation), so it makes sense for them to live in 'config' (you wouldn't want the 'api' package being read from the model layer.  I think it also doesn't make sense for api structs to live in the 'config' package.  
So for importing & organization purposes, i think it makes sense to duplicate these and have them be converted.

An alternative would be to basically do the same thing but in the DAO layer, but since these aren't really database related, i'm not sure that makes sense.

Reply:
And maybe a new packages makes more sense, open to ideas :)

Reply:
Just a thought, but what if they lived in config and were imported to api? 


```
// RepositoryParameterResponse holds data returned by a repositories API response
type RepositoryParameterResponse struct {
	config.DistributionVersions []DistributionVersion `json:"distribution_versions" ` //Versions available for repository creation
	config.DistributionArches   []DistributionArch    `json:"distribution_arches"`    //Architectures available for repository creation
}
```

Reply:
Not a bad idea, it would mean that we have json descriptors in the config package, but maybe that's better than having to copy the data from one to the other

Reply:
this is now done!